### DISCUSSION TITLE
How to change airfoil which is not in the list in CPACSCREATOR?

### DISCUSSION THREAD

**Discussion by kushaldgowda** [DLR-SL/CPACS#738] [Q&A] [answered]
How to change airfoil which is not in the list in CPACSCREATOR?
I tried with eclipse and edited x,y,z data. But I'm encountering so many error message from cpacscreator gui. 
Please upload instruction to edit code, or upload any tutorial video on that.

**Comment by MarAlder** [comment, marked as answer]
Using Eclipse, have you considered the correct arrangement of the profile points? It is described [in the documentation](https://www.cpacs.de/documentation/CPACS_3_3_0_Docs/html/a7e87ba7-5faf-7f61-cfd5-f5c0a96d1a9d.htm) as:

> Example 1: For a conventional wing, the airfoil coordinates are defined in x and z with all the y-coordinates set to "0". The points have to be ordered from the trailing edge along the lower side to the leading edge and then along the upper side back to the trailing edge.

Examples can be found here: 
https://github.com/DLR-SL/CPACS/blob/6cf5b21a3909b994050a8eb8fd53db8d2f620843/examples/basicWing.xml#L120-L128

**Reply by kushaldgowda** [reply]
I designed a wing with NACA 2315 in CPACS, saved, Then edited the co-ordinates of that in eclipse. and saved again. 
After that, It's showing these errors in cpacs like this.

ERR 11/02 11:37:09 TixiHelper.h:92] Failed to read element at xpath /cpacs/vehicles/aircraft/model/wings/wing[1]/segments/segment[1]: no profile algorithm

WRN 11/02 11:37:09 CPACSPositioning.cpp:68] Required element name is empty at xpath /cpacs/vehicles/aircraft/model/wings/wing[1]/positionings/positioning[2]

WRN 11/02 11:37:09 CPACSPositioning.cpp:111] Optional element fromSectionUID is present but empty at xpath /cpacs/vehicles/aircraft/model/wings/wing[1]/positionings/positioning[1]

WRN 11/02 11:37:09 CPACSPositioning.cpp:68] Required element name is empty at xpath /cpacs/vehicles/aircraft/model/wings/wing[1]/positionings/positioning[1]

ERR 11/02 11:37:09 CPACSProfileGeometry.cpp:109] Invalid choice configuration at xpath /cpacs/vehicles/profiles/wingAirfoils/wingAirfoil[2]

ERR 11/02 11:37:09 CPACSProfileGeometry.cpp:91] Failed to read pointList at xpath /cpacs/vehicles/profiles/wingAirfoils/wingAirfoil[2]: component vectors in CCPACSPointListXYZ must all have the same number of elements

**Reply by MarAlder** [reply]
Can you provide an example file? Looks to me like something is wrong in the CPACS file.

**Reply by kushaldgowda** [reply]
NACA2315 is reference file,
And NACA2315 modified is edited file. I replaced the co-ordinates here.
[Files.zip](https://github.com/DLR-SL/CPACS/files/7459406/Files.zip)

**Reply by MarAlder** [reply]
Try to remove a 0 from your y-vector. You 61 x values and 61 z values, but 62 y values.

**Reply by kushaldgowda** [reply]
Mann, Thanks, It worked 💯 

And now, Can I add that airfoil separately to the DBtemplate? (Profiles)
If I open new file after adding this airfoil to template, I can choose whatever I want based on presence of airfoil in template.

I mean in this file..

[profilesDB.zip](https://github.com/DLR-SL/CPACS/files/7459953/profilesDB.zip)

**Reply by MarAlder** [reply]
Nice to hear that it is working!

However, I don't really understand your last question. It should of course be possible to add your profile to the template. But I'm not really experienced with CpacsCreator - maybe @aidanjungo, @joergbrech or @rainman110 can help?

**Reply by kushaldgowda** [reply]
It worked in profile template aslo.  :)
